File: solutions/Javascript/Backspace String Compare.js
var backspaceCompare = function (s, t) {
    let s1 = [], s2 = [];
    for (let i = 0; i < s.length; i++) {
        if (s[i] == '#' && s[i].length > 0) {
            s1.pop();
        }
        else if (s[i] != '#') {
            s1.push(s[i]);
        }
    }
    for (let i = 0; i < t.length; i++) {
        if (t[i] == '#' && t[i].length > 0) {
            s2.pop();
        }
        else if (t[i] != "#") {
            s2.push(t[i]);
        }
    }
    return s1.join('') == s2.join('');
};

File: solutions/python/Backspace String Compare.py
class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        [s1,s2,i,j]=[[],[],0,0]
        while i<len(s) or j<len(t):
            if i<len(s) and  s[i] == '#' and len(s1)>0:
                s1.pop()
            if j<len(t) and t[j] == '#' and len(s2)>0:
                s2.pop()
            if i<len(s) and s[i] != '#':
                s1.append(s[i])
            if j<len(t) and t[i] !="#":
                s2.append(t[i])
            i += 1
            j += 1
        return ''.join(s1) == ''.join(s2)
            